### ISSUE TITLE
when deleting metrics graphite metrics, and they re-surface, send MetricData, not MetricPoint

### ISSUE THREAD

**Issue by Dieterbe** [raintank/tsdb-gw#143]
tsdbgw uses these 2 formats to send data into kafka for MT to consume:
* MetricData: full metric definition payloayd + timestamp and value (slow and bloaty)
* MetricPoint: optimized lightweight format only containing the id, timestamp and value.

tsdbgw typically first sends MetricData for a new metric, but from then on sends MetricPoint almost indefinitely, though occasionally it'll resend MetricData to be safe.

when a customer issues a delete (e.g. /metrics/delete which gets forwarded to MT) but keeps sending data, we should use MetricData again but the definition will be removed from the MT index and MT won't recognize the messages and drop the points as "unknown" (and trigger alerts)

possibly we may want to rethink index pruning as well in MT, to have some sort of feedback loop back to tsdbgw

**Comment by Dieterbe** [comment]
I jumped the gun a bit.
* as of #107 we regularly resend the full MetricData, to avoid this kind of problem. when users delete metrics but keep sending data, it's not unreasonable for the data to be unavailable for a while, until re-appearing a few hours later. but i just want to confirm the logic behaves as expected.
* pruning should not cause this issue, as by definition, pruning only happens for inactive metrics, not series that are still receiving data (but whether the cause is pruning or users issuing delete requests will be easily determined once our loki setup is in action)



likely we can just remove the alert for "unknown data" in MT and treat it as a normal occurence.

**Comment by Dieterbe** [comment]
I will do a test: delete some metrics and keep sending, and confirm they re-appear.
if that suceeds, can close this ticket and remove the alert.

**Comment by Dieterbe** [comment]
can confirm the earlier incident today was just normal activity: customer issuing deletes.

I did the test where I deleted a bunch of metrics in our ops environment.
unknown metrics in MT spikes up, then gradually builds down over the course of an hour
likewise, metrics are deleted and start becoming visible (including their past value, with a gap of a minute or two) over the course of the hour.
after an hour, all data shows fine and no more unknown metrics.

=> our internal alert should be, and will be removed.
=> there's no issue here.
